plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -46,6 +46,12 @@ class ProductCollection extends AbstractBlock {
 	 */
 	protected $custom_order_opts = array( 'popularity', 'rating' );
 
+	/**
+	 * Whether the render of the block should be prevented.
+	 *
+	 * @var bool
+	 */
+	private $prevent_render = null;
```

Comment:
@WunderBart Nitpick: TBH Code readability wise, I like the [idea](https://github.com/woocommerce/woocommerce/pull/50404#pullrequestreview-2227563244) of having separate states for `has_results` and `has_no_results_block`. Using `null` value for `prevent_render` is a bit confusing IMO. I would prefer to have separate states. This way, we can also easily avoid order of execution issues. FWIW I tried this out and you can see it [here](https://github.com/woocommerce/woocommerce/compare/fix/pc-prevent-rendering-on-empty-query...try/empty-pc?expand=1).

This is just my opinion. Feel free to go with your approach if you think it's better. I just wanted to share my thoughts on this. 🙂

Reply:
Separate states sound good - this way, it can be easily extended in case any additional variable comes into play. Thanks for the code, it’s actually solid and ready to be merged in! 😄 I’ll get back on it early next week (🆒👇)! 🙌

Reply:
I think it might make sense to have two props here, but for now, having one is also good in my opinion. I just wanted to suggest adding a hint in the docblock, saying that the props are runtime-based and get reinitialized for every block on a single page.

Reply:
Updated in 8ec5a8c02f0c9a20675b82cccacf7ffdef91b1d4